openedx/openedx-platform
# fix: Do not clear request cache when running eager Celery tasks

Description
When the CELERY_ALWAYS_EAGER setting variable is True celery tasks are run in the same thread as the Django Request. As a result, code that is intended to cleanup the celery worker inadvertently clears the RequestCache.

There are times in which we may want to run celery eagerly and still clear the cache, such as testing. I have set CLEAR_REQUEST_CACHE_ON_TASK_COMPLETION = False in all dev and test environments that already have CELERY_ALWAYS_EAGER = True. The unit test that specifically tests whether the request cache is getting cleared upon completion of a celery task then overrides CLEAR_REQUEST_CACHE_ON_TASK_COMPLETION = True even though CELERY_ALWAYS_EAGER = True for the sake of that specific testing purpose.

Supporting information
The issue and its resolution are discussed [here](https://discuss.openedx.org/t/extra-history-record-stored-on-each-problem-submission/8081/14).

Testing instructions
Run the platform using the devstack/test/bokchoy settings files, or simply defining CELERY_ALWAYS_EAGER = True. Confirm that the RequestCache is not cleared each time a celery task is triggered with apply_sync.

Deadline
None

Other information
None
